Networking/Networking/Remote/Remote.swift

```diff
@@ -346,12 +346,19 @@ private extension Remote {
 
 /// Contains the result of a paginated request.
 public struct PagedItems<T> {
+    /// Items fetched in this page
     public let items: [T]
+
+    /// Whether there are more pages after this one
     public let hasMorePages: Bool
 
-    public init(items: [T], hasMorePages: Bool) {
+    /// Number of items available, across all pages, whether loaded or not
+    public let totalItems: Int?
+
+    public init(items: [T], hasMorePages: Bool, totalItems: Int?) {
```

Comment:
Should we provide a nil default value here? Since (afaiu) this can only be nil due to decoding error now, we wouldn't need to pass `nil` explicitly in other places like `PointOfSaleCouponFetchStrategy` or mocks/previews:

```suggestion
    public init(items: [T], hasMorePages: Bool, totalItems: Int? = nil) {
```

Would also avoid having to deal with nil-coalescing in `totalItems: pagedProducts.totalItems ?? 0)` when tracking the fetch completion.